drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/CommonOnMoveHandler.java

```diff
@@ -0,0 +1,150 @@
+                /* It determines if the popover should be draw on the RIGHT or in the LEFT of the cell */
+                if (xPosition  + POPOVER_WIDTH > scenarioGrid.getLayer().getWidth()) {
+                    xPosition = (int) xYCell.getX() - xMiddleWidth;
+                    position = PopoverView.Position.LEFT;
+                }
+                /* Handling scrolling x-position */
+                int scrollX = scenarioGridPanel.getScrollPanel().getElement().getScrollLeft();
+                xPosition = xPosition - scrollX;
+                /* Handling scrolling y-position */
+                int yPosition = (int) xYCell.getY();
+                int scrollY = scenarioGridPanel.getScrollPanel().getElement().getScrollTop();
+                yPosition = yPosition - scrollY;
+                /* Parameters for the error message */
+                final Object expectedValue = factMappingValue.getRawValue();
+                final Object errorValue = factMappingValue.getErrorValue();
+                /* It shows the popover, the view depends on failed type (ERROR or EXCEPTION) */
+                if (FactMappingValueStatus.FAILED_WITH_ERROR == factMappingValue.getStatus()) {
+                    errorReportPopupPresenter.show(ScenarioSimulationEditorConstants.INSTANCE.errorReason(),
+                                                   ScenarioSimulationEditorConstants.INSTANCE.errorPopoverMessageFailedWithError(
+                                                           expectedValue != null ? expectedValue.toString() : NULL,
+                                                           errorValue  != null ? errorValue.toString() : NULL),
+                                                   ScenarioSimulationEditorConstants.INSTANCE.keep(),
+                                                   ScenarioSimulationEditorConstants.INSTANCE.apply(),
+                                                   () -> {
+                                                       scenarioGrid.getEventBus().fireEvent(
+                                                               new SetGridCellValueEvent(uiRowIndex,
+                                                                                         uiColumnIndex,
+                                                                                         errorValue != null ? errorValue.toString() : NULL));
+                                                   },
+                                                   xPosition,
+                                                   yPosition,
+                                                   position);
+                } else {
+                    errorReportPopupPresenter.show(ScenarioSimulationEditorConstants.INSTANCE.errorReason(),
+                                                   ScenarioSimulationEditorConstants.INSTANCE.errorPopoverMessageFailedWithException(
+                                                           factMappingValue.getExceptionMessage()),
```

Comment:
Do we need to check `null` value here as we do for `FAILED_WITH_ERROR` and if test is positive use constant `NULL`?

Reply:
Do you mean to test if <factMappingValue.getStatus()> is null?
Yes, I agree to add it, but in another point. In my opinion, if that value is NULL, the popover shouldn't be visible at all. I put it in line 87.